tests/unit_tests/rng/include/engines_api_tests.hpp

```diff
@@ -81,6 +94,19 @@ class engines_constructors_test {
 
 template <typename Engine>
 class engines_copy_test {
+
+    template <typename T>
+    void generate(oneapi::mkl::rng::bits<std::uint32_t> distr, T& engine, std::uint64_t n, cl::sycl::buffer<std::uint32_t, 1>& buf)
+    {
+        oneapi::mkl::rng::generate(distr, engine, n, buf);
+    }
+
+    template <>
+    void generate<oneapi::mkl::rng::mcg59>(oneapi::mkl::rng::bits<std::uint32_t> distr, oneapi::mkl::rng::mcg59 &engine, std::uint64_t n, cl::sycl::buffer<std::uint32_t, 1>& buf)
```

Comment:
Please, add a note, why we need to use n\2 here (it would be great if it's documented in the specification, but I am not sure)

Reply:
done